### PR TITLE
Update User Guide for v1.3

### PR DESCRIPTION
**Pull request by Effixion**
Still need to add screenshots where appropriate. Maybe we can also update what to add in the FAQ section.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -55,34 +73,83 @@ can get your contact and module management tasks done faster than traditional GU
...
+
+Examples:
+
+* `find CS2030S` returns `cs2030s`
+* `find bob jessica` returns `Bob McGhee` and `Jessica Lim`
+// Put screenshot here
+
+### Clearing all contacts and modules : `clear`
+
+Clears all entries from the contact list.
```
Comment (Yingming23) [inline docs/UserGuide.md:106]: Clear from both lists

File: docs/UserGuide.md [hunk 2]
```diff
@@ -97,96 +164,77 @@ Examples:
...
+* When editing assignment details, the existing assignment details of the module will be removed 
+  i.e adding of assignment details are not cumulative.
+* You can remove all the module’s assignment details by typing `a/` without specifying any assignment details after it.
 
 Examples:
 
-* `delete m 1`
+* `editm 1 l/Every Friday a/Functional Expressionism` Edits the lecture details and assignment details of the 1st module to be `Every Friday`
+  and `Functional Expressionism` respectively.
+* `editm 2 n/MA1521 t/` Edits the module code of the 2nd module to be `MA1521` and clears all existing assignment details.
```
Comment (Yingming23) [inline docs/UserGuide.md:222]: should be m/MA1521 and a/

File: docs/UserGuide.md [hunk 3]
```diff
@@ -97,96 +164,77 @@ Examples:
...
+* The module index refers to the index number shown in the displayed module list.
+* The index **must be a positive integer** 1, 2, 3, …​
 
-### Saving the data
+Examples:
 
-SOCompiler data are saved in the hard disk automatically after any command that changes the data. There is no need to
-save manually.
+* `list` followed by `deletem 2` deletes the 2nd module in the address book.
+* `find CS2030S` followed by `deletep=m 1` deletes the 1st module in the results of the `find` command.
```
Comment (Yingming23) [inline docs/UserGuide.md:237]: typo lol

File: docs/UserGuide.md [hunk 4]
```diff
@@ -200,25 +248,16 @@ the data of your previous SOCompiler home folder.
...
+| Add    | Module         | `addm m/MODULE_CODE [l/LECTURE_DETAILS] [t/TUTORIAL_DETAILS] [lz/LECTURE_ZOOM_LINK] [tz/TUTORIAL_ZOOM_LINK] [a/ASSIGNMENT_DETAILS]…​`          |
+| Add    | Contact        | `addp n/NAME [p/PHONE_NUMBER] [e/EMAIL] [m/MODULE_CODE] [t/TAG]…​`                                                                             |
+| Edit   | Module         | `editm INDEX [m/MODULE_CODE] [l/LECTURE_DETAILS] [t/TUTORIAL_DETAILS] [lz/LECTURE_ZOOM_LINK] [tz/TUTORIAL_ZOOM_LINK] [a/ASSIGNMENT_DETAILS]…​` |
+| Edit   | Contact        | `editp INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`                                                                               |
+| Delete | Module         | `deletem INDEX`                                                                                                                                |
+| Delete | Contact        | `deletec INDEX`                                                                                                                                |                                                                                                                                           |
```
Comment (Yingming23) [inline docs/UserGuide.md:258]: should be deletep

File: docs/UserGuide.md (hunk 2 above)
Comment (Effixion) [reply docs/UserGuide.md]: Fixed

File: docs/UserGuide.md (hunk 4 above)
Comment (Effixion) [reply docs/UserGuide.md]: Fixed

**Comment by Effixion** [comment]
Thanks Ying Ming for reviewing the UG

**Review by Yingming23 — APPROVED** [review]
LGTM

**Pull request merged**